lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Hunspell.java

```diff
@@ -49,20 +51,25 @@
  * shared by multiple spell-checkers in different threads).
  */
 public class Hunspell {
+  static final long SUGGEST_TIME_LIMIT = 250;
```

Comment:
I don't want to make this complicated, but for test purposes can we make this tweakable (even via package private method?)

I'm concerned about unit tests here being dependent on how fast the underlying system is, we don't want false failures. I have two concerns:
1) suggester tests unrelated (to this feature) that might sporadically fail because the system happens to be slow. 
2) suggester tests related to this feature that might sporadically fail because system is too fast.

There's a number of ways to ensure the tests stay robust. For example, in case 1, if we could set a TimeoutPolicy.NO_TIMEOUT (or similar name), we'd never hit false failures, even if say jenkins machine was running with load average in the hundreds. 

For case 2, where we really want to be 100% sure the timeout happens, even if it is on @mikemccand super-fast machine, it would be nice if we could set the limit to 0 (even if its a package-private setter just for testing or something).

Reply:
For case 1, NO_TIMEOUT looks like a great idea, thanks!
For case 2, maybe it can wait until so fast computers really appear?

Reply:
for the case 2, I just want to be extra paranoid and avoid any time-dependent tests. We've got battle scars here, we've had to wrestle with bugs in mac OS causing nanotime to go backwards. At my day job I recently spent a lot of time debugging such issues with hypervisors in jenkins (crazy jumps/stalls etc). It is very annoying to debug these issues even when code is doing the correct thing and using monotonic clock (like this PR).

So my point is the machine doesn't necessarily have to be crazy "fast" or "slow", especially with virtualization. Crazy stuff can and will happen, so if we can add extra caution for the tests it will be helpful in preventing flaky failures